localization/ekf_localizer/include/ekf_localizer/ekf_localizer.hpp

```diff
@@ -138,16 +153,18 @@ class EKFLocalizer : public rclcpp::Node
   };
 
   /* for model prediction */
-  geometry_msgs::msg::TwistStamped::SharedPtr
-    current_twist_ptr_;                                          //!< @brief current measured twist
-  geometry_msgs::msg::PoseStamped::SharedPtr current_pose_ptr_;  //!< @brief current measured pose
-  geometry_msgs::msg::PoseStamped current_ekf_pose_;             //!< @brief current estimated pose
+  std::queue<TwistInfo> current_twist_info_queue_;    //!< @brief current measured pose
+  std::queue<PoseInfo> current_pose_info_queue_;      //!< @brief current measured pose
```

Comment:
Did you try `std::deque`? It could be more useful for clear or index access methods.

Reply:
Yes, at first I used deque. 
But then I realized that queue is enough for the proposed implementation since I did not use `deque.push_front()` nor `deque.pop_back()`. Thus I thought queue would be better in this case. 

What do you think?